api/v1alpha1/inferencemodel_types.go

```diff
@@ -0,0 +1,168 @@
+	// modelName is often in reference to a LoRA adapter.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	TargetModels []TargetModel `json:"targetModels,omitempty"`
+	// Reference to the inference pool, the pool must exist in the same namespace.
+	//
+	// +kubebuilder:validation:Required
+	PoolRef *PoolObjectReference `json:"poolRef,omitempty"`
+}
+
+// PoolObjectReference identifies an API object within the namespace of the
+// referrer.
+type PoolObjectReference struct {
+	// Group is the group of the referent.
+	//
+	// +optional
+	// +kubebuilder:default="inference.networking.x-k8s.io"
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	Group string `json:"group,omitempty"`
+
+	// Kind is kind of the referent. For example "InferencePool".
+	//
+	// +optional
+	// +kubebuilder:default="InferencePool"
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
+	Kind string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Required
+	Name string `json:"name,omitempty"`
+}
+
+// Defines how important it is to serve the model compared to other models.
+// +kubebuilder:validation:Enum=Critical;Default;Sheddable
+type Criticality string
+
+const (
+	// Most important. Requests to this band will be shed last.
+	Critical Criticality = "Critical"
+	// More important than Sheddable, less important than Critical.
+	// Requests in this band will be shed before critical traffic.
+	// +kubebuilder:default=Default
+	Default Criticality = "Default"
+	// Least important. Requests to this band will be shed before all other bands.
+	Sheddable Criticality = "Sheddable"
+)
+
+// TargetModel represents a deployed model or a LoRA adapter. The
+// Name field is expected to match the name of the LoRA adapter
+// (or base model) as it is registered within the model server. Inference
+// Gateway assumes that the model exists on the model server and is the
+// responsibility of the user to validate a correct match. Should a model fail
+// to exist at request time, the error is processed by the Instance Gateway,
+// and then emitted on the appropriate InferenceModel object.
+type TargetModel struct {
+	// The name of the adapter as expected by the ModelServer.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// Weight is used to determine the proportion of traffic that should be
+	// sent to this target model when multiple versions of the model are specified.
+	//
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1000000
+	Weight int `json:"weight,omitempty"`
```

Comment:
Should be int32 per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

```suggestion
	Weight int32 `json:"weight,omitempty"`
```